Birthday-2024-Project/Scripts/UI/Puzzle/States/PuzzleUiWinState.gd

```diff
@@ -31,7 +31,7 @@ func enter_state():
 	
 	_ui_manager.show_main_screen()
 	
-	var screen = _ui_manager
+	var screen = _ui_manager.main_screen
 	screen.show_hide_win_text(true)
 	screen.context_button.text = "Next"
 	screen.context_button.button_up.connect(_on_next_clicked)
```

Comment:
Next, skip and reset buttons are leading to reload_current_scene which starts up GameManager _ready function. This way it realods main menu and is not working properly with puzzle. With this and Bishop's departure I have doubts about this state/manager solution

Reply:
I don't encounter any issues or errors when navigating Next, Skip, and Reset. It currently resets scene, but that's because those are just placeholder navigations.
Reset is implemented in the Level Editor PR.
Next and Skip have not been implemented yet, but that is more of a level select related task